### PR TITLE
upgrade kubernetes master/1.12 to bazel 0.16.1

### PR DESCRIPTION
**Pull request by BenTheElder**
we've done some testing and this appears to be stable again. we have had no breakage with bazel 0.15 or 0.16 per usual lately, except with remote caching, 0.16.1 contained a fix specifically for this and seems to be good.

/area images
/area jobs

### CODE REVIEW

File: experiment/bump_e2e_image.sh [hunk 1]
```diff
@@ -57,7 +57,6 @@ $SED -i "s/\\/kubekins-e2e:v.*-\\(.*\\)$/\\/kubekins-e2e:${TAG}-\\1/" "${TREE}/e
 bazel run //experiment:generate_tests -- \
   "--yaml-config-path=${TREE}/experiment/test_config.yaml" \
   "--output-dir=${TREE}/config/jobs/kubernetes/generated/"
-bazel run //jobs:config_sort
```
Comment (BenTheElder) [inline experiment/bump_e2e_image.sh]: fyi @krzyzacy

**Comment by ixdy** [comment]
/lgtm

Are you also pushing new images?

**Comment by BenTheElder** [comment]
Yeah the build was taking too long and then I had a meeting, sorry about that. They're pushed now

**Pull request merged**